akka-actor-typed-tests/src/test/java/jdocs/akka/typed/MutableIntroTest.java

```diff
@@ -138,4 +140,51 @@ public ChatRoomBehavior(ActorContext<RoomCommand> ctx) {
   }
   //#chatroom-actor
 
+
+  //#chatroom-gabbler
+  public static abstract class Gabbler {
+    private Gabbler() {
+    }
+
+    public static Behavior<ChatRoom.SessionEvent> behavior() {
+      return Behaviors.receive(ChatRoom.SessionEvent.class)
+          .onMessage(ChatRoom.SessionDenied.class, (ctx, msg) -> {
+            System.out.println("cannot start chat room session: " + msg.reason);
+            return Behaviors.stopped();
+          })
+          .onMessage(ChatRoom.SessionGranted.class, (ctx, msg) -> {
+            msg.handle.tell(new ChatRoom.PostMessage("Hello World!"));
+            return Behaviors.same();
+          })
+          .onMessage(ChatRoom.MessagePosted.class, (ctx, msg) -> {
+            System.out.println("message has been posted by '" +
+                msg.screenName +"': " + msg.message);
+            return Behaviors.stopped();
+          })
+          .build();
+    }
+
+  }
+  //#chatroom-gabbler
+
+  public static void runChatRoom() throws Exception {
+
+    //#chatroom-main
+    Behavior<Void> main = Behaviors.setup(ctx -> {
+      ActorRef<ChatRoom.RoomCommand> chatRoom =
+          ctx.spawn(ChatRoom.behavior(), "chatRoom");
+      ActorRef<ChatRoom.SessionEvent> gabbler =
+          ctx.spawn(Gabbler.behavior(), "gabbler");
+      ctx.watch(gabbler);
+      chatRoom.tell(new ChatRoom.GetSession("ol’ Gabbler", gabbler));
+
+      return Behaviors.receive(Void.class)
```

Comment:
`receive(Void.class)` looks a bit weird (even though it makes sense), I can expect newcomers stumble upon this. Any reason we can't have an `onSignal` factor method on `Behaviors` that returns a more constrained builder for only dealing with signals (or a `receiveSignals()` factory method)?

Reply:
Good catch, we can definitely. For example `Behaviors.<NotUsed>receiveSignal()`.

I think this was left from before we had all the current factories.

Reply:
Actually closest currently possible is: 
```
return Behaviors.<Void>receiveSignal(
  (c, sig) -> {
    if (sig instanceof Terminated) return Behaviors.stopped();
    else return Behaviors.unhandled();
  }
);
```
Which is somewhat verbose but I don't think the case of specifying an actor that does not accept any message is common and useful enough to warrant special builder API.

Reply:
That's fine. But what I meant was: ``return Behaviors.<Void>receiveSignal().onSignal(...).onSignal()...``

Reply:
I think that earlier API design experience has told us that Java devs are not very used to explicitly passing specifying a type parameter on a method like that, and are more comfortable with an actual parameter, so for example we have `Flow.of(Element.class)` for Java while the Scala side has `Flow[Element]`.

Except for that we could surely add a `Behaviors.<T>receive()` returning a builder (because you may want to define it and _not_ accept any signals either), but then maybe it is better to have just one way to do it than two similar with no obvious benefit of one over the other?